tortoise/expressions.py

```diff
@@ -1,32 +1,67 @@
-from typing import Union
+from typing import TYPE_CHECKING, Optional, Tuple, Type, Union
 
 from pypika import Field
 from pypika.terms import ArithmeticExpression
 
 from tortoise.exceptions import FieldError
 
+if TYPE_CHECKING:  # pragma: nocoverage
+    from tortoise.models import Model
+
 
 class F(Field):  # type: ignore
     @classmethod
     def resolver_arithmetic_expression(
         cls,
-        fields_db_projection: dict,
+        model: "Type[Model]",
         arithmetic_expression_or_field: Union[ArithmeticExpression, Field],
-    ):
+    ) -> Tuple[Union[ArithmeticExpression, Field], Optional[Field]]:
+        field_object = None
+
         if isinstance(arithmetic_expression_or_field, Field):
             name = arithmetic_expression_or_field.name
             try:
-                name = fields_db_projection[name]
+                arithmetic_expression_or_field.name = model._meta.fields_db_projection[name]
+
+                field_object = model._meta.fields_map.get(name, None)
+                if field_object:
+                    func = field_object.get_for_dialect(
+                        model._meta.db.capabilities.dialect, "function_cast"
+                    )
+                    if func:
+                        arithmetic_expression_or_field = func(
+                            field_object, arithmetic_expression_or_field
+                        )
             except KeyError:
-                raise FieldError(f"Field {name} is virtual and can not be updated")
-            arithmetic_expression_or_field.name = name
+                raise FieldError(
+                    f"Field {name} is not default field in model,"
+                    f" so can not be used with F expression"
```

Comment:
The message here is confusing to me? Do you say this only works with fields that have a default set?

Reply:
Yeah. Maybe this message can confuse user. Can you recommend more clearer message?

Reply:
We'll only get a `KeyError` if the field is not a DB column. So maybe we should explicitly mention that?
`There is no DB-backed field called {name}`?
or
`There is no non-virtual field {name} on Model {model.__name__}`?

Reply:
I think below looks better ! I will change the message :+1: